include/simple/generic_socket.hpp

```diff
@@ -99,22 +106,21 @@ class GenericSocket
     int data_past_topic{0};
     auto data_past_topic_size{sizeof(data_past_topic)};
 
-    zmq_msg_t message{};
-    zmq_msg_init(&message);
+    zmq_msg_init(&recv_message_);
```

Comment:
This will clean the current message even if we don't receive anything, use a local zmq_msg and move it to the member one only if a receive was successful